can/io/asc.py

```diff
@@ -381,8 +390,8 @@ def on_message_received(self, msg: Message) -> None:
                 symbolic_name="",
                 brs=1 if msg.bitrate_switch else 0,
                 esi=1 if msg.error_state_indicator else 0,
-                dlc=msg.dlc,
-                data_length=len(data),
+                dlc=len2dlc(msg.dlc),
```

Comment:
Yeah, `msg.dlc` is confusing (#1212). This change and the one to `test/logformats_test.py` looks good.